### PR TITLE
TunableOp size hotfix

### LINKED ISSUE #130727: Encountering HIP OOM error for the baddbmm operator when auto-tune is enabled
### 🐛 Describe the bug

When using the below environment variables to enable the auto-tune mode for the baddbmm operator (with input tensors with certain strides) on AMD GPUs:

`PYTORCH_TUNABLEOP_ENABLED=1 PYTORCH_TUNABLEOP_TUNING=1 PYTORCH_TUNABLEOP_FILENAME="hipblas_tuning.csv" PYTORCH_TUNABLEOP_MAX_TUNING_DURATION_MS=30 PYTORCH_TUNABLEOP_MAX_WARMUP_DURATION_MS=30 HIP_FORCE_DEV_KERNARG=1 PYTORCH_TUNABLEOP_VERBOSE=1`

The program will encounter HIP OOM error. A mini repo is in the below:

```
import torch

def main() -> None:
    gpu_device = torch.device("cuda")

    input_tensor = torch.rand(
        (1920, 1, 100),
        device=gpu_device,
        dtype=torch.bfloat16,
    )
    input_tensor = torch.as_strided(
        input_tensor, size=(1920, 1, 100), stride=(100, 100, 1)
    )
    batch1_tensor = torch.rand(
        (1920, 256, 512),
        device=gpu_device,
        dtype=torch.bfloat16,
    )
    batch1_tensor = torch.as_strided(
        batch1_tensor, size=(1920, 256, 512), stride=(512, 983040, 1)
    )
    batch2_tensor = torch.rand(
        (1920, 512, 100),
        device=gpu_device,
        dtype=torch.bfloat16,
    )
    batch2_tensor = torch.as_strided(
        batch2_tensor, size=(1920, 512, 100), stride=(51200, 100, 1)
    )

    for _ in range(100):
        _ = torch.baddbmm(input_tensor, batch1_tensor, batch2_tensor)


if __name__ == "__main__":
    main()
```

The below is the example error message:
```
_ = torch.baddbmm(input_tensor, batch1_tensor, batch2_tensor)
  torch.OutOfMemoryError: HIP out of memory. Tried to allocate 900.00 GiB. GPU 0 has a total capacity of 191.98 GiB of which 186.38 GiB is free. Of the allocated memory 1.02 GiB is allocated by PyTorch, and 1.63 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_HIP_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)
```

### Versions

pytorch 2.5.0

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang

### PR DESCRIPTION
**Pull request by jeffdaily**
Fixes #130727.  GetSize calculation was incorrect for strided batched gemm.

### CODE REVIEW

File: aten/src/ATen/cuda/tunable/GemmCommon.h [hunk 1]
```diff
@@ -85,15 +85,15 @@ struct GemmParams : OpParams {
   }
 
   size_t GetSizeA() const {
-    return sizeof(T) * lda * ((transa == 'n' || transa == 'N') ? k : m);
+    return sizeof(T) * m * k;
```
Comment (xw285cornell) [inline aten/src/ATen/cuda/tunable/GemmCommon.h:88]: noob q, we don't need to worry about transpose any more?

Comment (jeffdaily) [reply aten/src/ATen/cuda/tunable/GemmCommon.h:88]: This fix was too brash.  Expect something better.  We're writing more unit tests with this PR to capture all cases and this change here breaks when you have a 4D tensor sliced down to 2D.

Comment (jeffdaily) [reply aten/src/ATen/cuda/tunable/GemmCommon.h]: Resolved.  The original size calcs were correct for standard gemm.  Only the strided batched was wrong.  Updated, with a new unit test to cover.  PR is minimal diff now.

**Review by xw285cornell — APPROVED** [review]
LGTM!

**Pull request closed without merging**